### PR TITLE
Develop optional rule args

### PR DESCRIPTION
**Pull request by GiaJordan**
With the introduction of rules: `list`, `protectAges`, `unique`, and `inRange`, there are now multiple rules that allow arguments but do not require all of them to be specified every time because default behavior is defined. 

The process of declaring novel rules available to the suite has been simplified. The three dictionaries `valid_rule_combinations`, `validation_types` ( in the `validate_rules_utils.py`), and the separate `validation_types` ( in the `validate_manifest.py`) have been combined into the `validation_rule_info` dict that holds a dict of information for each rule containing arguments, type, and complementary rules available for use. New rules can now be declared in a single place, and rule spec updates are contained to one function in one file.

This PR also closes the bug in #675 where rules that required 1 argument and were missing it were not flagged as invalid.

### CODE REVIEW

**Review by milen-sage — APPROVED** [review]
Tentatively approved. But let's wait until the next release of schematic before merging.

**Comment by GiaJordan** [comment]
@milen-sage will merge after release this week

**Comment by GiaJordan** [comment]
Merging now that release 22.8.1 is live

**Pull request merged**
